src/System.Net.WebHeaderCollection/src/System/Net/HeaderInfoTable.cs

```diff
@@ -3,88 +3,152 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections;
+using System.Collections.Generic;
 using System.Collections.Specialized;
+using System.Linq;
+using System.Text;
 
 namespace System.Net
 {
     internal class HeaderInfoTable
     {
-        private static Hashtable HeaderHashTable;
-        private static HeaderInfo UnknownHeaderInfo = new HeaderInfo(string.Empty, false, false, false);
+        private static readonly Func<string, string[]> s_singleParser = value => new[] { value };
+        private static readonly Func<string, string[]> s_multiParser = value => ParseValueHelper(value, isSetCookie: false);
+        private static readonly Func<string, string[]> s_setCookieParser = value => ParseValueHelper(value, isSetCookie: true);
+        private static readonly HeaderInfo s_unknownHeaderInfo = new HeaderInfo(string.Empty, false, false, false, s_singleParser);
+        private static readonly Hashtable s_headerHashTable;
+
+        private static string[] ParseValueHelper(string value, bool isSetCookie)
+        {
+            // RFC 6265: (for Set-Cookie header)
+            // If the name-value-pair string lacks a %x3D ("=") character, ignore the set-cookie-string entirely.
+            if (isSetCookie && (value.IndexOf('=') < 0)) return Array.Empty<string>();
+
+            var tempStringCollection = new List<string>();
+
+            bool inquote = false;
+            int startIndex = 0;
+            int length = 0;
+
+            for (int i = 0; i < value.Length; i++)
+            {
+                if (value[i] == '\"')
+                {
+                    inquote = !inquote;
+                }
+                else if ((value[i] == ',') && !inquote)
+                {
+                    if (!isSetCookie || !IsDuringExpiresAttributeParsing(value.SubstringTrim(startIndex, length)))
```

Comment:
Ideally the string would only be allocated once, e.g.:

```c#
string singleValue = value.SubstringTrim(startIndex, length);
if (!isSetCookie || !IsDuringExpiresAttributeParsing(singleValue))
{
    tempStringCollection.Add(singleValue);
    startIndex = i + 1;
    length = 0;
    continue;
}
```

Reply:
Will fix.